shared/steps/setup/merge_pr_git_repository.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package setup
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"testing"
+
+	"github.com/open-component-model/ocm-e2e-framework/shared"
+)
+
+type Payload struct {
+	Do                     string `json:"Do"`
+	MergeMessageField      string `json:"MergeMessageField"`
+	MergeTitleField        string `json:"MergeTitleField"`
+	MergeWhenChecksSucceed bool   `json:"merge_when_checks_succeed"`
+	ForceMerge             bool   `json:"force_merge"`
+}
+
+// MergePullRequest merges a PR
+func MergePullRequest(repoName string, prNumber int) features.Func {
+	return func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
+		t.Helper()
+
+		data := Payload{
+			Do:                     "squash",
+			MergeMessageField:      "PR Merge",
+			MergeTitleField:        "Auto Merge PR with after successful Validation Check",
+			MergeWhenChecksSucceed: false,
+			ForceMerge:             true,
+		}
+		payloadBytes, err := json.Marshal(data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		body := bytes.NewReader(payloadBytes)
+		url := fmt.Sprintf("%s/api/v1/repos/%s/%s/pulls/%d/merge", shared.BaseURL, shared.Owner, repoName, prNumber)
```

Comment:
Why not use the gitea API here?

Reply:
I'm asking why not use the gitea client API to run a MergePullRequest instead of constructing the authentication and body by hand?

Reply:
The git sdk does not provide these 2 options:

`MergeWhenChecksSucceed: false, 
ForceMerge:             true,`

If you remember, the SDK was throwing a 405 because the checks were taking a longer time to succeed than the speed of the test. We would have to add a wait over here, which would have added unnecessary complexity as it has to be a sleep a or something and not something graceful like the wait.For from the k8s e2e framework

Reply:
Ah right. Nice. Perhaps add a little note why you're doing it. 👍